src/components/UpsertMovieModal.js

```diff
@@ -1,10 +1,29 @@
 import "./UpsertMovieModal.css";
 import React from "react";
-import { useForm } from "react-hook-form";
 import GenresRepo from "../services/GenreRepo";
 import { useDispatch } from "react-redux";
 import { bindActionCreators } from "redux";
 import { actionCreators } from "../store";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+
+const MovieSchema = Yup.object().shape({
+    title: Yup.string().required(),
+    url: Yup.string().required(),
+    releaseDate: Yup.string().required(),
+    rating: Yup.number()
```

Comment:
For number you can use `min()` and `max()`
```suggestion
    rating: Yup.number(),min(0).max(10)
```